docs/data/data-grid/components/components.md

```diff
@@ -153,22 +123,13 @@ As any component slot, every icon can be customized. However, it is not yet poss
 
 See the [Overlays](/x/react-data-grid/overlays/) documentation on how to customize the `loadingOverlay`, `noRowsOverlay`, and `noResultsOverlay`.
 
-## Slot props
-
-To override default props or pass custom props to slot components, use the `slotProps` prop.
+## Custom slot props with TypeScript
```

Comment:
I kept the module augmentation doc here for now because only the grid has it so it felt weird to movie it fully to the shared page.